From 2aad107b9df5d254392505d3a90c33643de5db40 Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Mon, 12 Feb 2024 10:51:37 +0100
Subject: [PATCH 0792/1695] drivers: stm32_remoteproc: disable trustzone on low
 power exit

When resuming from low power exit, we need to disable trustzone
as it is enabled by default.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1f35ee5bd5ae9b02209421e1e2c7ae0213b40e9d drivers: stm32_remoteproc: disable trustzone on low power exit]
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/359356
Change-Id: Ia6c3ff9270b50654b69edc081dacf2189eba20d9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/375028
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 core/drivers/remoteproc/stm32_remoteproc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/drivers/remoteproc/stm32_remoteproc.c b/core/drivers/remoteproc/stm32_remoteproc.c
index efb234edf..1d3749f5f 100644
--- a/core/drivers/remoteproc/stm32_remoteproc.c
+++ b/core/drivers/remoteproc/stm32_remoteproc.c
@@ -511,6 +511,10 @@ static void stm32_rproc_a35ss_cfg(struct stm32_rproc_instance *rproc __unused)
 	stm32mp_syscfg_write(A35SSC_M33CFG_ACCESS_CR, rproc->m33_cr_right,
 			     A35SSC_M33_TZEN_CR_M33CFG_SEC |
 			     A35SSC_M33_TZEN_CR_M33CFG_PRIV);
+	/* Disable the TrustZone that is enabled by default */
+	stm32mp_syscfg_write(A35SSC_M33_TZEN_CR, 0,
+			     A35SSC_M33_TZEN_CR_CFG_SECEXT);
+
 #endif /* defined(CFG_STM32MP25) || defined(CFG_STM32MP23) */
 }
 
-- 
2.47.2

